src/internal/mod.rs

```diff
@@ -838,11 +885,40 @@ fn gen_plaintext_and_aug_with_retry<R: CryptoOps>(
     aug_private_key().or_else(|_| aug_private_key())
 }
 
+/// Runs a future with a timeout or just runs the future, depending on if a timeout is specified.
+///
+/// If a timeout limit is reached, the result will be an IronOxideErr::OperationTimedOut.
+/// If no timeout is specified, or if the operation finishes before the timeout, the
+/// result is the result of the sdk operation.
+pub async fn run_maybe_timed_sdk_op<F: Future>(
```

Comment:
This doesn't run anything, it just adds timeout to it. Name suggestion `add_future_timeout` or `add_timeout` and let the fact that it takes a future do the talking.

Reply:
Went with `add_optional_timeout` -- seem better?